features/core/src/main/java/org/apache/karaf/features/internal/service/FeatureConfigInstaller.java

```diff
@@ -137,10 +137,10 @@ public void installFeatureConfigs(Feature feature) throws IOException, InvalidSy
                     cfg = createConfiguration(configAdmin, cid);
                     cfgProps.put(CONFIG_KEY, cid.pid);
                     properties.put(CONFIG_KEY, cid.pid);
+                    cfg.update(cfgProps);
```

Comment:
Don't we want to keep the update there ?

Reply:
I moved the update up, so it's updated _without_ `felix.fileinstall.filename` property at this stage - this ensures that fileinstall (handling `CM_UPDATED`) won't read+write the file being written at the same time with `updateStorage()`